### DISCUSSION TITLE
Problem creating a NITF TRE with the nitf.jni-c.dll

### DISCUSSION THREAD

**Discussion by jeffdugan** [mdaus/nitro#227] [Q&A]
Hi All,
We have been using a Java interface to create nitf, 2.7 version, files from KLVs for some time in a glassfish/mule environment.
We have recently taken the processing out of the glassfish/mule paradigm and created a java process that relies on ActiveMQ for handling messages/commands. Everything works fine except for a portion of the code that attempts to create a nitf file from some KLV data. The jars and dll included in the newly deployed lib directory all seem to be the same as the previous deployment.
I can tell that the nitf.jni-c.dll is being loaded b/c if I change the name of the dll temporarily I get the UnsatisfiedLinkError error output. 
I also tried defining the nitf_plugin_path to a share/nitf/plugins directory that also contains another copy of the nitf.jni-c.dll.
When I do this I get and error "[While retrieving DLL hook] (..\c\nrt\source\DLLWin32.c, 122, unknown function): ' failed to get function [nitf.jni-c_init] from dll [{path}\nitf.jni-c.dll]' : Invalid hook in DSO.
This error seems to comes from the c portion of code that is trying to create the PluginRegistry object.
If I run with and w/o defining the nitf_plugin_path I additionally get the "NITF Exception: TRE cannot be found for this TRE: SENSRA."
Any suggestions are greatly appreciated.
Thanks,
Jeff

**Comment by JDanielSmith** [comment]
From the `README.md`: *The MATLAB and JAVA layers have not been touched in years. Use at your own risk.*

**Comment by jeffdugan** [comment]
I saw that disclaimer, but, was hoping someone in the forum may be aware of what is happening and may have a helpful suggestion.
